code/ATMOSPHERICS/atmospherics.dm

```diff
@@ -21,9 +21,19 @@ Pipelines + Other Objects -> Pipe network
 	var/pipe_color
 	var/obj/item/pipe/stored
 
+	var/welded = 0 //Used on pumps and scrubbers
+
 	var/global/list/iconsetids = list()
 	var/global/list/pipeimages = list()
 
+
+/obj/machinery/atmospherics/Destroy()
+	for(var/mob/living/L in src)
+		L.remove_ventcrawl()
+		L.loc = get_turf(src)

```

Comment:
One thing I just realised is that this doesn't called the Enter procs.
You might want to use forceMove ( or whatever the TG equivalent is )